src/app/(proper_react)/redesign/(authenticated)/user/dashboard/Dashboard.test.tsx

```diff
@@ -543,55 +638,919 @@ it("shows scan in progress indicators on the dashboard with results and unresolv
   expect(exposureTableDescription).toBeInTheDocument();
 });
 
-it("shows scan in progress indicators on the dashboard with resolved scan results and unresolved breaches", () => {
+// Check dashboard banner content for story DashboardNonUsNoBreaches
+it("shows the correct dashboard banner title for non-US users, no breaches", () => {
+  const ComposedDashboard = composeStory(DashboardNonUsNoBreaches, Meta);
+  render(<ComposedDashboard />);
+
+  const dashboardTopBannerTitle = screen.queryAllByText("No exposures found");
+  expect(
+    getRelevantBannerContentElement(dashboardTopBannerTitle)
```

Comment:
I'm not sure I understand why `getRelevantBannerContentElement` is needed? Is there no way to get the relevant elements in a way that does [not rely on invisible class names](https://testing-library.com/docs/queries/about/#priority)? (And if there is a good reason, could that be added to the JSDoc of `getRelevantBannerContentElement`?)

Reply:
My thinking was the following: I’d like to be able to get and query the dashboard top banner elements as usual. It’s possible for other elements to match the same text, label and roles outside of the top banner. Adding test IDs to the elements that are being tested would get cucumbersome. That’s why I decided to add `getRelevantBannerContentElement` as a helper to filter out elements that should be not taken into account for certain tests.

Added a comment in [84fd600](https://github.com/mozilla/blurts-server/pull/3494/commits/84fd600c51683baf38f79ccd77aa0116d3ea14a4). Do you have a suggestion on how this could be improved?

Reply:
Ah, OK! Yes, that comment clarifies a lot. (Also, sorry, when I mentioned "why it's needed" in the overall review, I was referring to this specifically, not to the tests themselves :sweat_smile: )

However, this actually points out an area where we're currently lacking in accessibility. One way people with e.g. low vision or low dexterity navigate a page, is by first listing the [landmarks](https://developer.mozilla.org/en-US/blog/aria-accessibility-html-landmark-roles/) to determine the overall areas of the page, and then navigating within those landmarks - almost exactly what you're trying to do in these tests. But this top section isn't currently a labelled landmark (and neither is the main content below).

So, if you want to do The Right Thing™, I'd suggest wrapping the top banner in a `<section>` that either has an `aria-labelledby` pointing to the `<h3>` in there (which is a bit of a pain due to how the top banner is split up, and also I'm not sure how descriptive that heading is of its contents anyway), or add a direct `aria-label`. And then the area below the top banner would be the `<main>` part, I think.

You can then select the banner itself using

```tsx
  const banner = screen.getByRole("region", { name: "<the heading or aria-label content>" })
```

and then select elements _inside_ the banner using the generic (i.e. not a property on `screen`) `getByRole` function, with `banner` as its first element:

```tsx
  const dashboardTopBannerCta = getByRole(banner, "link", {
    name: "Monitor more emails",
  });
```

If you feel like that might be too much work and not worth the effort, let me know, then I'll approve anyway, but at least will also have made you feel bad :P (And possibly I'll make the change myself in a followup PR.)

Reply:
That is a good point and I’m in favor of doing The Right Thing™. Since the top banners not being accessible is not due to recent changes I’ll follow up on this. The unit tests are valuable for making sure that the changes introduced in #3493 don’t break and it makes debugging them easier we should land them as soon as possible.